### PR TITLE
Feature: Add torchserve logger

### PR DESCRIPTION
**Pull request by jagadeeshi2i**
This PR adds logger support for torchserve.

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:
```release-note
Add logger support for torchserve.
```

### CODE REVIEW

**Review by chauhang — APPROVED** [review]

File: docs/samples/custom/torchserve/logger/logger_demo.ipynb [hunk 1]
```diff
@@ -0,0 +1,364 @@
+{
```
Comment (yuzisun) [inline docs/samples/custom/torchserve/logger/logger_demo.ipynb:1]: <p>should the dumper deployed in the same namespace as inference service  in kfserving-test namespace?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/kubeflow/kfserving/pull/1228/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

Comment (yuzisun) [inline docs/samples/custom/torchserve/logger/logger_demo.ipynb:1]: <p>port and component-port is not exposed on the yaml spec, we can actually fix the logger port to use an uncommon port number </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/kubeflow/kfserving/pull/1228/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

File: docs/samples/custom/torchserve/logger/README.md [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    minReplicas: 1
+    logger:
+      url: http://message-dumper.default/
+      mode: all
+    containers:
+      - image: {username}/torchserve:latest
+        name: torchserve-container
+```
+
+* By default the logger port is 8081 and component port is 8080. Run torchserve inference on 8080 and management on a port other than 8081.
```
Comment (yuzisun) [inline docs/samples/custom/torchserve/logger/README.md:42]: I have fixed this now, it uses the uncommon port https://github.com/kubeflow/kfserving/blob/master/cmd/agent/main.go#L37

Comment (jagadeeshi2i) [reply docs/samples/custom/torchserve/logger/README.md:42]: @yuzisun is it possible to have it configurable as torchserve with kfserving wrapper will not use 8080 port for inference.

File: docs/samples/custom/torchserve/logger/logger_demo.ipynb [hunk 3]
```diff
@@ -0,0 +1,364 @@
...
+      "\u001b[94mkind\u001b[39;49;00m: \u001b[33m\"\u001b[39;49;00m\u001b[33mInferenceService\u001b[39;49;00m\u001b[33m\"\u001b[39;49;00m\r\n",
+      "\u001b[94mmetadata\u001b[39;49;00m:\r\n",
+      "  \u001b[94mname\u001b[39;49;00m: \u001b[33m\"\u001b[39;49;00m\u001b[33mtorchserve-custom\u001b[39;49;00m\u001b[33m\"\u001b[39;49;00m\r\n",
+      "\u001b[94mspec\u001b[39;49;00m:\r\n",
+      "  \u001b[94mpredictor\u001b[39;49;00m:\r\n",
+      "    \u001b[94mminReplicas\u001b[39;49;00m: 1      \r\n",
+      "    \u001b[94mlogger\u001b[39;49;00m:\r\n",
+      "      \u001b[94murl\u001b[39;49;00m: http://message-dumper.default/\r\n",
+      "      \u001b[94mport\u001b[39;49;00m: 8091\r\n",
+      "      \u001b[94mcomponent-port\u001b[39;49;00m: 8083\r\n",
```
Comment (yuzisun) [inline docs/samples/custom/torchserve/logger/logger_demo.ipynb:89]: Can you help remove these two fields which are not supported?

**Pull request merged**